Documentation/guides/OneDotNetEmbeddedResources.md

```diff
@@ -0,0 +1,280 @@
+
+    lib
+        net6.0-android29\Foo.dll
+        net6.0-android30\Foo.dll
+    # Optional reference assemblies
+    ref
+        net6.0-android29\Foo.dll
+        net6.0-android30\Foo.dll
+    # Optional native libraries
+    runtimes
+        android-arm\native\libFoo.so
+        android-arm64\native\libFoo.so
+        android-x86\native\libFoo.so
+        android-x64\native\libFoo.so
+
+There is not a great place where all Android file types would fit
+following this pattern.
+
+In Android Studio, Android libraries are packaged as [`.aar`][aar]
+files. A Xamarin.Android library, `Foo.csproj`, could also generate an
+[`.aar`][aar] file in its `$(OutputPath)`:
+
+    Foo.aar
+        classes.jar
+        res/
+        assets/
+        libs/*.jar
+        jni/[arch]/*.so
+
+`@(AndroidEnvironment)` files are specific to Xamarin.Android, so we
+can make one addition to the file format:
+
+    Foo.aar
+        env/MyEnvironment.txt
+
+So the output of `Foo.csproj` would look like:
+
+    bin/Debug/[targetframework]/
+        Foo.dll
+        Foo.aar
+    bin/Release/[targetframework]/
+        Foo.dll
+        Foo.aar
+
+If you ran the `Pack` target, you would get a `.nupkg` file with:
+
+    lib
+        net6.0-android29\Foo.dll
+        net6.0-android29\Foo.aar
+        net6.0-android30\Foo.dll
+        net6.0-android30\Foo.aar
+
+When consuming the `.nupkg` files, Xamarin.Android will still have to
+extract [`.aar`][aar] files on disk for command-line tooling such as
+`javac`, `d8/r8` and `manifestmerger`. If users want to copy around
+loose build output and consume it, there will only be 1 additional
+[`.aar`][aar] file they will need to copy.
+
+### Java/Kotlin Dependencies
+
+A `Foo.csproj` might include `bar.aar` and `baz.jar` files that are
+Java/Kotlin dependencies.
+
+`.aar` files should sit alongside the .NET assembly:
+
+    lib
+        net6.0-android29\Foo.dll
+        net6.0-android29\Foo.aar
+        net6.0-android29\bar.aar
+        net6.0-android30\Foo.dll
+        net6.0-android30\Foo.aar
+        net6.0-android30\bar.aar
+
+The `baz.jar` file would be included in the above `Foo.aar` file at:
+
+    Foo.aar
+        classes.jar
+        libs/baz.jar
+
+### Native Libraries
+
+Since both `.aar` and `.nupkg` files support native libraries,
+Xamarin.Android should support consuming native libraries from both
+locations. A Xamarin.Android class library, `Foo.csproj` will place
+native libraries in the `Foo.aar` file by default.
+
+### Assembly-Level Attributes
+
+To mark each `.aar` file that is needed for an assembly, a new custom
+attribute is needed:
+
+```csharp
+[assembly: Android.Runtime.Dependency ("Foo.aar")]
+[assembly: Android.Runtime.Dependency ("bar.aar")]
+```
+
+We can emit a build error if any files are missing.
+
+We also need to emit an attribute to mark the assembly as using the
+new system:
+
+```csharp
+[assembly: Android.Runtime.Features (Android.Runtime.Feature.NativeDependencies)]
```

Comment:
Let's please use `Java.Interop` and not `Android.Runtime`, and it can live in `Java.Interop.dll`.